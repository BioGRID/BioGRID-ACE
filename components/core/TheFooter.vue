<template>
    <div class="the-footer">
        <v-footer
            app
            color="primary"
            inset
            :fixed="false"
        >
            <v-container class="pa-0">
                <v-row>
                    <v-col
                        class="text-center white--text"
                        cols="12"
                    >
                        <span>Copyright &copy; {{ new Date().getFullYear() }}, <a :href="copyrightLinkout" :title="copyrightOwner">{{ copyrightOwner }}</a>, All Rights Reserved</span>
                    </v-col>
                </v-row>
            </v-container>
        </v-footer>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'nuxt-property-decorator'

@Component
export default class TheFooter extends Vue {
    private copyrightOwner: string = 'BioGRID Project'
    private copyrightLinkout: string = 'https://thebiogrid.org'
}
</script>

<style lang="scss" scoped>
     a, a:visited {
         color: #ffcc2d;
         font-weight: bold;
         text-decoration: none;
     }
     a:hover {
         color: #cc0910;
     }
</style>
