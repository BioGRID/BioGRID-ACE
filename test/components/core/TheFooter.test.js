
// LIBRARIES
import Vuetify from 'vuetify'
import Vuex from 'vuex'

// COMPONENTS
import TheFooter from '@/components/core/TheFooter.vue'

// UTILITIES
import { shallowMount, createLocalVue } from '@vue/test-utils'

const localVue = createLocalVue()
localVue.use(Vuex)

let wrapper

beforeEach(() => {
    const vuetify = new Vuetify()
    wrapper = shallowMount(TheFooter, {
        localVue,
        vuetify
    })
})

afterEach(() => {
    wrapper.destroy()
})

describe('TheFooter', () => {
    test('renders', () => {
        expect(wrapper.exists()).toBe(true)
    })
    test('renders properly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})
