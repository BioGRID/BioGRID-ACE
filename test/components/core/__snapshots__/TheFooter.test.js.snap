// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/core/TheFooter renders properly 1`] = `
"<div class=\\"the-footer\\">
  <v-footer-stub color=\\"primary\\" height=\\"auto\\" tag=\\"footer\\" app=\\"true\\" inset=\\"true\\">
    <v-container-stub tag=\\"div\\" class=\\"pa-0\\">
      <v-row-stub tag=\\"div\\">
        <v-col-stub cols=\\"12\\" tag=\\"div\\" class=\\"text-center white--text\\"><span>Copyright Â© 2021, <a href=\\"https://thebiogrid.org\\" title=\\"BioGRID Project\\">BioGRID Project</a>, All Rights Reserved</span></v-col-stub>
      </v-row-stub>
    </v-container-stub>
  </v-footer-stub>
</div>"
`;
